src/status_im/mailserver/core.cljs

```diff
@@ -575,10 +601,14 @@
   (when (contains? db :multiaccount)
     (let [last-connection-attempt (:mailserver/last-connection-attempt db)]
       (when (and (fetch-use-mailservers? cofx)
-                 (<= (- now last-connection-attempt)))
-        (fx/merge cofx
-                  (when (not= :connected (:mailserver/state db))
-                    (change-mailserver)))))))
+                 ;; We are not connected
+                 (not= :connected (:mailserver/state db))
+                 ;; We either never tried to connect to this peer
+                 (or (nil? last-connection-attempt)
+                     ;; Or 30 seconds have passed and no luck
+                     (<= (- now last-connection-attempt) (/ (* constants/connection-timeout 3) 1000))))
```

Comment:
now in ms ? why we don't divide it as well ?

Reply:
you are right, we should not divide it at all, nice catch!